pkg/utils/apiutil/serverapi/middleware.go

```diff
@@ -233,3 +234,38 @@ func (h *redirector) ServeHTTP(w http.ResponseWriter, r *http.Request, next http
 	client := h.s.GetHTTPClient()
 	apiutil.NewCustomReverseProxies(client, urls).ServeHTTP(w, r)
 }
+
+const (
+	backoffMaxDelay = 3 * time.Second
```

Comment:
LGTM! a nitpick :) 

Do we need to consider changing this time if the client sets a timeout for the request?
such as `&http.Client{Timeout: 2 * time.Second}`